flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcFilterPushdownPreparedStatementVisitor.java

```diff
@@ -0,0 +1,178 @@
+            return renderBinaryOperator(">", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.GREATER_THAN_OR_EQUAL.equals(call.getFunctionDefinition())) {
+            return renderBinaryOperator(">=", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.OR.equals(call.getFunctionDefinition())) {
+            return renderBinaryOperator("OR", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.AND.equals(call.getFunctionDefinition())) {
+            return renderBinaryOperator("AND", call.getResolvedChildren());
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<ParameterizedPredicate> renderBinaryOperator(
+            String operator, List<ResolvedExpression> allOperands) {
+        Optional<ParameterizedPredicate> leftOperandString = allOperands.get(0).accept(this);
+
+        Supplier<Optional<ParameterizedPredicate>> rightOperandString =
+                () -> allOperands.get(1).accept(this);
+
+        return leftOperandString.flatMap(
+                left -> rightOperandString.get().map(right -> left.combine(operator, right)));
+    }
+
+    @Override
+    public Optional<ParameterizedPredicate> visit(ValueLiteralExpression litExp) {
+        LogicalType tpe = litExp.getOutputDataType().getLogicalType();
+        Class<?> typeCs = tpe.getClass();
+
+        if (supportedDataTypes.contains(typeCs)) {
+            ParameterizedPredicate predicate = new ParameterizedPredicate("?");
+
+            Serializable[] params = new Serializable[1];
+
+            if (typeCs.equals(VarCharType.class)) {
+                params[0] = litExp.getValueAs(String.class).orElse(null);
+            }
+            if (typeCs.equals(BigIntType.class)) {
+                params[0] = litExp.getValueAs(Long.class).orElse(null);
+            }
+            if (typeCs.equals(IntType.class) || typeCs.equals(SmallIntType.class)) {
+                params[0] = litExp.getValueAs(Integer.class).orElse(null);
+            }
+            if (typeCs.equals(DoubleType.class)) {
+                params[0] = litExp.getValueAs(Double.class).orElse(null);
+            }
+            if (typeCs.equals(BooleanType.class)) {
+                params[0] = litExp.getValueAs(Boolean.class).orElse(null);
+            }
+            if (typeCs.equals(FloatType.class)) {
+                params[0] = litExp.getValueAs(Float.class).orElse(null);
+            }
+            if (typeCs.equals(DecimalType.class)) {
+                params[0] = litExp.getValueAs(BigDecimal.class).orElse(null);
+            }
+            if (typeCs.equals(DateType.class)) {
+                params[0] = litExp.getValueAs(LocalDate.class).map(Date::valueOf).orElse(null);
+            }
+            if (typeCs.equals(TimestampType.class)) {
+                params[0] =
+                        litExp.getValueAs(LocalDateTime.class).map(Timestamp::valueOf).orElse(null);
```

Comment:
Timestamps might be tricky. What if one database accepts `yyyy-MM-dd'T'HH:mm:ss.SSSZ` while other one requires `yyyy-MM-dd HH:mm:ss.SSSZ` (note there is no `'T'`)? The format might differ per dialect.